### PR TITLE
docs(firebase_messaging): Updated Firebase_messaging documentation

### PR DESCRIPTION
**Pull request by MAkcanca**
## Description

Changed the registerWith function in Firebase_messaging documentation, suggested as [this](https://github.com/FirebaseExtended/flutterfire/issues/1904#issuecomment-580868406).

## Related Issues

(#4396)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by Salakar** [comment]
:wave what version of FlutterFire messaging did you test this on? The docs website is for the v8 dev release which has a completely different Android package name/structure so this change won't work;

image

File: docs/messaging/overview.mdx [hunk 1]
```diff
@@ -57,7 +57,7 @@ public class Application extends FlutterApplication implements PluginRegistrantC
 
   @Override
   public void registerWith(PluginRegistry registry) {
-    GeneratedPluginRegistrant.registerWith(registry);
+    FirebaseMessagingPlugin.registerWith(registry.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"));
```
Comment (rrousselGit) [inline docs/messaging/overview.mdx:60]: Why such change? With GeneratedPluginRegistrant you shouldn't need to do that

Comment (Salakar) [reply docs/messaging/overview.mdx:60]: Yep, and even then it's `FlutterFirebaseMessagingPlugin`, FirebaseMessagingPlugin doesn't exist

Comment (MAkcanca) [reply docs/messaging/overview.mdx:60]: This was written in the documentation itself, I just implemented it and pushed it, nothing new. Maybe implementation changed, I will check this out.

**Pull request closed without merging**